### PR TITLE
add metric for excluded apps

### PR DESCRIPTION
**Pull request by mcleinman**
## Description

Adding session metric for count of excluded apps. Ping is sent in https://github.com/mozilla-mobile/mozilla-vpn-client/pull/6303.

Data review was [previously approved](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5702#pullrequestreview-1272682462).

Also improving glean categorization for a prior metric.

## Reference

https://mozilla-hub.atlassian.net/browse/VPN-4066

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

File: src/apps/vpn/apppermission.cpp [hunk 1]
```diff
@@ -103,6 +103,12 @@ QVariant AppPermission::data(const QModelIndex& index, int role) const {
   }
 }
 
+int AppPermission::disabledAppCount() {
+  SettingsHolder* settingsHolder = SettingsHolder::instance();
+  QStringList applist = settingsHolder->vpnDisabledApps();
```
Comment (brizental) [inline src/apps/vpn/apppermission.cpp:108]: nit: This variable is unnecesary.

**Review by brizental — CHANGES_REQUESTED** [review]
Actually. Please add a unit test for this new colection.

File: qtglean/metrics.yaml [hunk 2]
```diff
@@ -24,6 +24,7 @@ performance:
     expires: never
     time_unit: microsecond
 
+settings:
```
[the remark this answers is not in the document]
Comment (mcleinman) [reply qtglean/metrics.yaml:27]: Good call. I uplifted a separate PR that used the `settings` header to 2.15.0, so nothing will go live to production and then change.

**Comment by mcleinman** [comment]
> Actually. Please add a unit test for this new colection.

A unit test of "record a metric glean, then check that it recorded it"? I believe we ran into some issues w/ trying to do better functional tests before, but can do something like this if you'd like.

**Comment by brizental** [comment]
> A unit test of "record a metric glean, then check that it recorded it"? I believe we ran into some issues w/ trying to do better functional tests before, but can do something like this if you'd like.

I meant to add a unit test... I see the Controller doesn't have any, though. I have filed https://mozilla-hub.atlassian.net/browse/VPN-4445 to look into it. Please file a follow up for us to write tests for this collection, that is important. You can block it on the controller ticket I just filed and on https://mozilla-hub.atlassian.net/browse/VPN-4149. I'll make sure to prioritize VPN-4149 on the next sprint.

**Review by brizental — APPROVED** [review]

**Comment by mcleinman** [comment]
> > A unit test of "record a metric glean, then check that it recorded it"? I believe we ran into some issues w/ trying to do better functional tests before, but can do something like this if you'd like.
> 
> I meant to add a unit test... I see the Controller doesn't have any, though. I have filed https://mozilla-hub.atlassian.net/browse/VPN-4445 to look into it. Please file a follow up for us to write tests for this collection, that is important. You can block it on the controller ticket I just filed and on https://mozilla-hub.atlassian.net/browse/VPN-4149. I'll make sure to prioritize VPN-4149 on the next sprint.

Ticket created: https://mozilla-hub.atlassian.net/browse/VPN-4452

**Pull request merged**